### ISSUE TITLE
Improve responsiveness when there is background noise

### ISSUE THREAD

**Issue by darrenlucas** [tomcollins/bbc-voice#15]
Currently the speech recognition seems to wait until the user doesn't say anything further in order to process the input.  Change this to either:
* Continuously gather input text and parse until a phrase match is found
* Continuously gather input text up until a time and / or word limit is reached

Set the interimResults property to true on the webkitSpeechRecognition object: https://dvcs.w3.org/hg/speech-api/raw-file/tip/speechapi.html#dfn-interimresults

**Comment by owainlewis** [comment]
I think we can continually poll for input like you suggested. Was thinking this the other day actually.

**Comment by owainlewis** [comment]
The only thing that makes this difficult is determining the END or terminal input token. I.e at what point do we stop. Like the idea of "when a phrase match is found". That could work. But the free form nature of the speech might make it hard. Worth thinking about.

**Comment by owainlewis** [comment]
Perhaps

1. When "BBC" activation word happens we start processing.
2. After 2-3 seconds we stop processing regardless and see if we have a match.
3. If the word BBC happens again restart to process (in case you get half way through nothing happens and you just want to start again)

**Comment by darrenlucas** [comment]
Yes, or after 1 or 2 seconds of not receiving a new word then stop processing.

It would also be useful to have visual feedback that the keyword has activated, and also when it has stopped.  Saying the keyword should also probably stop and tasks in progress, e.g the speech synthesis reading of the headlines.

**Comment by owainlewis** [comment]
Like the idea of visual feedback. Almost like a "listening..." type thing. Might make it feel more reactive. (i.e sometimes you don't know if it's doing anything and it just hangs there). I'm still not sold on the idea of auto reading things any more. (well on news. It works really well on weather) as it just trips up the voice stuff too much.
